cpp/src/io/parquet/legacy/parquet_reader_impl.cu

```diff
@@ -659,8 +659,8 @@ table reader::Impl::read(int skip_rows, int num_rows, int row_group) {
                                 : col_meta.data_page_offset;
         const auto buffer =
             source_->get_buffer(offset, col_meta.total_compressed_size);
-        page_data.emplace_back(buffer->data(), align_size(buffer->size()));
-        d_compdata = static_cast<uint8_t *>(page_data.back().data());
+        page_data[chunks.size()] = rmm::device_buffer(buffer->data(), buffer->size());
```

Comment:
I'm having trouble understanding the relationship of `num_column_chunks` to `chunks.size()`.  `page_data` has size `num_column_chunks` -- just making sure this is guaranteed: `chunks.size() < num_column_chunks`.

Reply:
Yes, it is increasingly adding column chunks (up to num_column_chunks immediately afterwards, so the current size  is always the index of the column chunk being added (the immediately following hostdevice_vector::insert() call is really a push_back)